### PR TITLE
[doc] Update PULL_REQUEST_TEMPLATE.md

### PR DESCRIPTION
**Pull request by saimidu**
*Issue #, if available:*

## PR Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [x] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [x] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [x] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [x] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

## Reviewer Checklist
* For reviewer, before merging, please cross-check:
- [x] I've verified the code change on the config file mentioned above has already been reverted

*Description:*
Add a new check to the checklist which asks contributors to add compulsory pytest markers to any new test functions they may have added.

*Tests run:*
N/A

*DLC image/dockerfile:*
N/A

*Additional context:*
N/A

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 - [ ] (If applicable) I've documented below the tests I've run on the DLC image
 - [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
 - [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.
+- [ ] (If applicable) I've added the markers `@pytest.mark.integration("<feature-being-tested>")` and `@pytest.mark.model("<model-type>")` to the new tests which I have added
```
Comment (arjkesh) [inline .github/PULL_REQUEST_TEMPLATE.md:9]: model is the only required marker I believe. if you are going to add integration, can you also add `and @pytest.mark.multinode(num_nodes) if the test runs on more than one instance`

Comment (saimidu) [reply .github/PULL_REQUEST_TEMPLATE.md]: Added. Please take a look at the updated template.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,12 +1,20 @@
...
+- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
+- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
+- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"inf">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
+- [ ] (If applicable) I have added the marker `@pytest.mark.canary("<reason>")` to the new tests which I have added, if a test should be run as a canary test apart from being run as just a unit/integration test
+- [ ] (If applicable) I have added the marker `@pytest.mark.skipif(is_pr_context(), reason="<reason-for-skip>")` to the new tests which I have added, if a test must not be run on Pull Request CI tests
+
```
Comment (arjkesh) [inline .github/PULL_REQUEST_TEMPLATE.md:17]: please remove these

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,12 +1,20 @@
...
 - [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.
 
+## Pytest Marker Checklist
+- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
+- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
+- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
+- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"inf">)` to the new tests which I have added, if a test is specifically applicable to only one processor type
```
Comment (arjkesh) [inline .github/PULL_REQUEST_TEMPLATE.md:14]: please change inf to "neuron"

**Review by arjkesh — APPROVED** [review]

**Pull request merged**